bcdaworker/main.go

```diff
@@ -293,5 +295,61 @@ func main() {
 	fmt.Println("Starting bcdaworker...")
 	workerPool := setupQueue()
 	defer workerPool.Close()
+	expiredScheduler()
 	waitForSig()
 }
+
+func updateStatus(hrThreshold int) {
```

Comment:
Consider that there will be multiple instances of this app running across many hosts. What happens if this routine runs at the same time across said hosts? Will this introduce a race condition?

Reply:
Yes I think, at the very least, lines 313, 325, and 334 are susceptible to a race condition between workers.  It may be best to leverage a system-wide cron instead of an app-specific cron.

Reply:
@rnagle:
Good point, definitely something I should have thought about. 

@msnook, to the list below:
1. Yes!! A system level cron is where it's at. Using Ryan's approach for the cron tab in combination with a system level cron you suggest would handle the race condition issue.

2. The timestamp part was already implemented so we're good there. Implemented lines 321 - 323
3. That's in the code, line 331 - 334.

Reply:
+1 for system level cron.  As long as we run it on only one machine it seems fine to me.